### PR TITLE
Add Global Vars widget, dialog, context menus

### LINKED ISSUE #3042: Ability to define global variables
In the disassembly view of a program,  I see instructions like `mov rax, qword [0x00005040]`, where `0x00005040` is a global variable in the .data segment. I couldn't find a way to actually create a global variable at that location though. In the same way we can define right click and define a function, we should be able to right click and define an address as data.

Ideally though, Cutter should be able to automatically define a variable here for me, as this is what Ghidra, IDA, and other programs do.

### PR DESCRIPTION
**Pull request by XVilka**
# SQAUSH ME

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [ ] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**

- [x] Add Global Vars widget
- [x] Add "Add Global Var" action in the Disassembly context menu (just like "Add Flag ...")
- [x] Add a context menu to Add/Edit/Delete global var in the Global Vars widget

**Test plan (required)**

- Open any file
- Add global vars (you can use `avga` and `avg` commands)
- Refresh the widget contents and observe them
- Add/modify global vars by right-clicking on the address, current or as operand, in the Disassembly or Graph widgets, see "Add at..." submenu
- Check these widgets: Globals, Disassembly, Graph, Decompiler - they should update automatically

**Closing issues**

Closes https://github.com/rizinorg/cutter/issues/3042

**Screenshots**

Screenshot 2023-07-03 at 22 44 26

Screenshot 2023-08-02 at 01 33 12

Screenshot 2023-08-04 at 01 24 28

### CODE REVIEW

File: src/core/Cutter.cpp [hunk 1]
```diff
@@ -4022,6 +4048,99 @@ QList<XrefDescription> CutterCore::getXRefs(RVA addr, bool to, bool whole_functi
...
 
+void CutterCore::addGlobalVariable(RVA offset, QString name, QString typ)
+{
+    name = sanitizeStringForCommand(name);
+    CORE_LOCK();
+	char *errmsg = NULL;
+	RzType *globType = rz_type_parse_string_single(core->analysis->typedb->parser,
+			typ.toStdString().c_str(), &errmsg);
+	if (errmsg) {
+		qWarning() << tr("Error parsing type: \"") << typ << tr("\" message: ") << errmsg;
```
Comment (wargio) [inline src/core/Cutter.cpp]:
```suggestion
		qWarning() << tr("Error parsing type: \"%1\" message: ").arg(typ) << errmsg;
```
same for all the others.

File: src/core/Cutter.cpp [hunk 2]
```diff
@@ -4022,6 +4048,99 @@ QList<XrefDescription> CutterCore::getXRefs(RVA addr, bool to, bool whole_functi
...
+		free(errmsg);
+		return;
+	}
+	if (!rz_analysis_var_global_create(core->analysis, name.toStdString().c_str(),
+				globType, offset)) {
+		qWarning() << tr("Error creating global variable: \"") << name << tr("\"");
+		return;
+	}
+
```
Comment (wargio) [inline src/core/Cutter.cpp]: formatting. run clang-format.

Comment (XVilka) [reply src/core/Cutter.cpp]: CI doesn't check formatting in reality, I updated it in a separate PR: https://github.com/rizinorg/cutter/pull/3216

File: src/menus/DisassemblyContextMenu.cpp [hunk 3]
```diff
@@ -166,6 +166,19 @@ QWidget *DisassemblyContextMenu::parentForDialog()
...
 
+void DisassemblyContextMenu::addAddAtMenu()
+{
+    setAsMenu = addMenu(tr("Add at..."));
+
+    initAction(&actionRename, tr("Rename or add flag"), SLOT(on_actionRename_triggered()),
+               getRenameSequence());
+    setAsMenu->addAction(&actionRename);
+
+    initAction(&actionGlobalVar, tr("Change or add global variable"),
```
Comment (wargio) [inline src/menus/DisassemblyContextMenu.cpp]: I prefer modify
```suggestion
    initAction(&actionGlobalVar, tr("Modify or add global variable"),
```

**Review by wargio — APPROVED** [review]

**Pull request merged**